spot/spot_code/Generate_Spot_Annotations.py

```diff
@@ -0,0 +1,260 @@
+                slide_compressed_counts['main_cell_types'] = pd.concat([slide_compressed_counts['main_cell_types'], added_df], axis=1)
+
+                # Add these into pct_states and pct_subtypes?
+
+        slide_compressed_counts['main_cell_types'] = slide_compressed_counts['main_cell_types'].sort_index(axis=1)
+        return slide_compressed_counts
+
+    def process_spots(self):
+
+        # Initializing annotations using wak
+        spot_annotations = wak.Annotation(mpp=self.mpp)
+        spot_annotations.add_names(['Spots'])
+
+        # Iterating through barcodes and creating equal sized spots centered on coordinates
+        spot_pixel_diameter = int((1/self.mpp)*55)
+        spot_pixel_radius = int(spot_pixel_diameter/2)
+        for b in tqdm(self.barcodes):
+
+            # Pulling out that row from coordinates
+            b_coords = self.coordinates.loc[b]
+            b_x = b_coords['imagecol']
+            b_y = b_coords['imagerow']
+            
+            # Creating shapely annotation for this spot
+            spot_poly = Point(b_x, b_y).buffer(spot_pixel_radius)
+
+            # Extracting -omics information from previous processing step
+            spot_main_cell_types = self.omics_data['main_cell_types'].loc[b].to_frame()
+            spot_main_cell_types.columns = ['Main_Cell_Types']
+            spot_main_cell_types_dict = spot_main_cell_types.to_dict()
+
+            # Cell State info
+            spot_cell_state_dict = {}
+            spot_cell_subtype_dict = {}
+            for m in list(spot_main_cell_types_dict['Main_Cell_Types'].keys()):
+
+                spot_cell_state = self.omics_data[m]['pct_states'][b].to_frame()
+                spot_cell_state.columns = [m]
+                spot_cell_state_dict[m] = spot_cell_state.to_dict()[m]
+
+                spot_cell_subs = self.omics_data[m]['pct_subtypes'][b].to_frame()
+                spot_cell_subs.columns = [m]
+                spot_cell_subs_dict = spot_cell_subs.to_dict()
+                spot_cell_subtype_dict[m] = spot_cell_subs_dict[m]
+
+            spot_properties = {
+                'Main_Cell_Types': spot_main_cell_types_dict['Main_Cell_Types'],
+                'Cell_States': spot_cell_state_dict,
+                'Cell_Subtypes': spot_cell_subtype_dict
+                }
+            # Adding spot to annotations, crs is just the origin since we are using non-scaled centroid points
+            # name is set to be the barcode
+            spot_annotations.add_shape(
+                poly=spot_poly,
+                box_crs=[0, 0],
+                structure='Spots',
+                name=b,
+                properties=spot_properties)
+
+        # Print str property of Annotation object
+        print(spot_annotations)
+        return spot_annotations
+
+    def save(self):
+
+        os.makedirs(self.args.basedir + '/tmp/')
+        f = open(self.args.basedir + '/tmp/spotanot.json', 'w')
+        f.close()
```

Comment:
Is it necessary to write the annotations to a tmp directory prior to posting? It looks like the annotations are created, written, read, then dumped and posted.

Reply:
Made changes so that json file will not be saved.